lib/components/app-service-bot.js

```diff
@@ -6,10 +6,16 @@
  * @param {MatrixClient} client The client instance configured for the AS bot.
  * @param {AppServiceRegistration} registration The registration that the bot
  * is following. Used to determine which user IDs it is controlling.
+ * @param {Function=} setJoined Called when a /joined_members request has a remote user.
```

Comment:
In the future we might want to consider another controller class. Having the language generic would be a good thing to in the event that /joined_members gets replaced. The arguments to said function also need documenting.

And finally, what's the use case for this?